tests/core/pool.c

```diff
@@ -32,26 +25,24 @@ void test_core_pool__1(void)
 	int i;
 	git_pool p;
 
-	cl_git_pass(git_pool_init(&p, 1, 4000));
+	git_pool_init(&p, 1);
+	p.page_size = 4000;
 
 	for (i = 2010; i > 0; i--)
 		cl_assert(git_pool_malloc(&p, i) != NULL);
 
 	/* with fixed page size, allocation must end up with these values */
-	cl_assert_equal_i(1, git_pool__open_pages(&p));
-	cl_assert_equal_i(507, git_pool__full_pages(&p));
-
+	cl_assert_equal_i(590, git_pool__open_pages(&p));
```

Comment:
@carlosmn: No, this is because the test is exercising the pathological case for the slab allocator. It creates a 2010 byte object, and then creates another one which is just 2009 bytes, so it falls *just* outside the available space for the current page, and we need to allocate a new one. It ends up with a lot of pages that are roughly 50% full, but without a free-list, the half-full pages cannot be reused like before.

In real world scenarios, we don't allocate objects that are 50% the size of a page, so the waste is nowhere as large.